Tests/DataStreamTests.swift

```diff
@@ -0,0 +1,749 @@
+//
+//  DataStreamTests.swift
+//
+//  Copyright (c) 2020 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Alamofire
+import XCTest
+
+final class DataStreamTests: BaseTestCase {
+    func testThatDataCanBeStreamedOnMainQueue() {
+        // Given
+        let expectedSize = 1000
+        var accumulatedData = Data()
+        var response: HTTPURLResponse?
+        var streamOnMain = false
+        var completeOnMain = false
+        let expect = expectation(description: "stream should complete")
+
+        // When
+        AF.streamRequest(URLRequest.makeHTTPBinRequest(path: "bytes/\(expectedSize)")).responseStream { stream in
+            switch stream.event {
+            case let .stream(result):
+                streamOnMain = Thread.isMainThread
```

Comment:
So far the test data isn't so big that multiple calls are needed. I could request more data, but since the handling of the stream side is unified, a single check should be all that's needed. I can add more if you think it's important.

Reply:
No that's fine. Just thought it may be warranted. Sounds like it's not.